packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -345,13 +366,14 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
                 )
               }
             }
-            chunk.code = s.toString()
-            // TODO source map
+            updateChunkCode(s)
           }
 
           // there may still be markers due to inlined dynamic imports, remove
           // all the markers regardless
-          chunk.code = chunk.code.replace(preloadMarkerRE, 'void 0')
+          const s = new MagicString(chunk.code)
```

Comment:
Could we create this MagicString at the top so we only use one, and only need to combine the source maps once?

Reply:
It needs to replace markers which were not replaced. Using the same MagicString instance was doing a different behavior here.
```js
import MagicString from 'magic-string';

const mark = '__VITE_PRELOAD__';

const code = `
  import('foo.js', "__VITE_PRELOAD__")
  "__VITE_PRELOAD__"
`;

const s = new MagicString(code);

s.overwrite(
  code.indexOf(mark) - 1,
  code.indexOf(mark) + mark.length + 1,
  '["deps.js"]',
  {
    contentOnly: true,
  }
);

/*
  import('foo.js', ["deps.js"])
  "__VITE_PRELOAD__"
*/
console.log(s.toString());

s.replace(new RegExp(`"${mark}"`, 'g'), 'void 0');

/*
  import('foo.js', void 0)
  void 0
*/
console.log(s.toString());
```

To change this, it will need to store positions at line 355 and then do the replace without `s.replace` at line 375 with skipping the stored positions.
Should I do so? (It's going to be easier than I first thought.)

Reply:
> To change this, it will need to store positions at line 355 and then do the replace without `s.replace` at line 375 with skipping the stored positions.
> Should I do so? (It's going to be easier than I first thought.)

Sounds like a good thing to do now that I re-look the code, should improve perf too.

Reply:
I have changed it 👍